scripts/utils/user.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+set -exuo pipefail
+
+####################################################################
+# Setup user and groups for inside the container
+
+# It seems that clashes with group names or GIDs is more common
+# than one might think. Here we attempt to make a matching group
+# inside the container, but if it fails, we abandon the attempt.
+
+# Try to create the group to match the GID. If a group already exists
+# with that name, but a different GID, no change will be made.
+# We therefore run groupmod to ensure the GID does match what was
+# requested.
+# However, either of these steps could fail - but if they do, 
+# that's OK.
+groupadd -fg "${GID}" "${GROUP}" || true
+groupmod -g "${GID}" "${GROUP}" || true
+
+group_info=$(getent group "$GROUP")
+group_info=(${group_info//:/ })
```

Comment:
This is clever but maybe a little too much so.  I think it is more idiomatic to just set IFS and let the shell split the fields for you.

```
$ IFS=: g=($(getent group users))
$ echo ${g[2]}
100
```